submissions/MosQuitO404/oop-exercise/index.js

```diff
@@ -0,0 +1,90 @@
+import { print } from './js/lib.js';
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/MosQuitO404/a-tiny-JS-world
+   Web app: https://mosquito404.github.io/a-tiny-JS-world/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Inhabitant {
+   constructor(species, name, gender, legs, saying) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.legs = legs;
+      this.saying = saying;
+      this.properties = ['species', 'name', 'gender', 'legs', 'saying'];
+   }
+   
+   get showProperties() {
+      return this.properties.map(property => this[property]).join('; ');
+   }
+}
+
+class Animal extends Inhabitant {
+   constructor(species, name, gender, saying) {
+      super(species, name, gender, 4, saying);
+   }
+}
+
+class Dog extends Animal {
+   constructor(name, gender) {
+      super('Dog', name, gender, 'Woof-woof!');
+   }
+}
+
+class Cat extends Animal {
+   constructor(name, gender) {
+      super('Cat', name, gender, 'Meow-meow!');
+   }
+}
+
+class Human extends Inhabitant {
+   constructor(name, gender, saying) {
+      super('Human', name, gender, 2, saying);
+      this.hands = 2;
+      this.properties = ['species', 'name', 'gender', 'legs', 'hands', 'saying'];
```

Comment:
A child class shouldn't handle base class' properties unless there are good reasons to break encapsulation principle.
There is no good reasons for this in this task.
Each class should restrict properties handling to the properties they own.
Look into method overloading and calling inherited versions of methods via `super` (original project docs offer links to relevant materials).

Reply:
Done!